arviz/rcparams.py

```diff
@@ -93,11 +102,64 @@ def _validate_boolean(value):
     return value is True or value == "true"
 
 
+def iterable_validator(validate_scalar, length=None):
+    """Validate value is an iterable datatype."""
+    # no blank lines allowed after function docstring by pydocstyle,
+    # but black requires white line before function
+
+    def validate_iterable(value):
+        if isinstance(value, str):
+            try:
+                val = [validate_scalar(v.strip("([ ])")) for v in value.split(",") if v.strip()]
+                if length is not None:
+                    if len(val) == length:
+                        return val
+                    else:
+                        raise ValueError("Iterable must be of length: {}".format(length))
+                else:
+                    return val
+            except Exception:
+                raise
+        elif np.iterable(value) and not isinstance(value, (set, frozenset)):
+            val = [validate_scalar(v) for v in value]
+            if length is not None:
+                if len(val) == length:
+                    return val
+                else:
+                    raise ValueError("Iterable must be of length: {}".format(length))
+            else:
+                return val
+        else:
+            raise ValueError("Only iterable values are valid")
+
+    return validate_iterable
+
+
+def _make_validate_float_iterable(length):
+    """Validate value is a float iterable of given length."""
+    # no blank lines allowed after function docstring by pydocstyle,
+    # but black requires white line before function
+
+    def validate_float_iterable(value):
+        if value is None or isinstance(value, str) and value.lower() == "none":
+            return None
+        elif value == "auto":
+            return value
+        try:
+            validate_floatlist = iterable_validator(_validate_float_or_none, length)
+            return tuple(validate_floatlist(value))
```

Comment:
Instead of converting to tuple here, it is probably better to make `validate_iterable` always return tuples. Otherwise, the following pattern is possible:

```
In [2]: plt.rcParams["lines.dashdot_pattern"]
Out[2]: [6.4, 1.6, 1.0, 1.6]

In [3]: plt.rcParams["lines.dashdot_pattern"].append("unvalidated_value")

In [4]: plt.rcParams["lines.dashdot_pattern"]
Out[4]: [6.4, 1.6, 1.0, 1.6, 'unvalidated_value']
```

And I do not think it should be. It allows to set rcParams to unvalidated values which eventually makes `plt.plot(x,y , "-.")` fail with a weird error message

Reply:
Sorry, I don't understand. Can you please elaborate how unvalidated values are allowed?

Reply:
The main difference between a python list and a tuple is that lists are mutable (it can be modified after creation, is elements can be changed, or it can have new elements appended) whereas tuples are not. Therefore, in the above example, the append is modifying the list instead of modifying the `lines.dashdot_pattern` rcParam which is what runs the validation function when setting values. 

Note that in the example we have ended up with a string in the list, whereas trying to set this same list via rcParams (with the validation function) will give the following output:

```
In [7]: plt.rcParams["lines.dashdot_pattern"] = [6.4, 1.6, 1.0, 1.6, 'unvalidated_value']
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/venvs/random/lib/python3.6/site-packages/matplotlib/rcsetup.py in __call__(self, s)
    318                     else val
--> 319                     for val in s]
    320         except ValueError:

~/venvs/random/lib/python3.6/site-packages/matplotlib/rcsetup.py in <listcomp>(.0)
    318                     else val
--> 319                     for val in s]
    320         except ValueError:

ValueError: could not convert string to float: 'unvalidated_value'

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
~/venvs/random/lib/python3.6/site-packages/matplotlib/__init__.py in __setitem__(self, key, val)
    801             try:
--> 802                 cval = self.validate[key](val)
    803             except ValueError as ve:

~/venvs/random/lib/python3.6/site-packages/matplotlib/rcsetup.py in __call__(self, s)
    320         except ValueError:
--> 321             raise ValueError('Could not convert all entries to floats')
    322 

ValueError: Could not convert all entries to floats

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-7-cb71fbf16588> in <module>
----> 1 plt.rcParams["lines.dashdot_pattern"] = [6.4, 1.6, 1.0, 1.6, 'unvalidated_value']

~/venvs/random/lib/python3.6/site-packages/matplotlib/__init__.py in __setitem__(self, key, val)
    802                 cval = self.validate[key](val)
    803             except ValueError as ve:
--> 804                 raise ValueError("Key %s: %s" % (key, str(ve)))
    805             dict.__setitem__(self, key, cval)
    806         except KeyError:

ValueError: Key lines.dashdot_pattern: Could not convert all entries to floats
```

Reply:
If tuples were used, no values can be appended, so the only way of updating it is to create a new tuple with the new element and make `rcParams["key"] = new_tuple` which will then always be validated

Reply:
Got it, I'll make the changes!